.github/workflows/update-schema.yml

```diff
@@ -1,40 +1,95 @@
-name: update provider schema
+name: enforce schema consistency
 
 on:
-  push:
-    branches:
-      - main
+  pull_request:
     paths:
       - 'internal/**'
       - 'pkg/**'
       - 'go.mod'
       - 'go.sum'
-  workflow_dispatch: {}
-
-permissions:
-  contents: write
 
 jobs:
-  update-schema:
-    name: update provider schema
+  enforce-schema-consistency:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      - uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+
+      - name: Set up Go
+        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
         with:
           go-version-file: go.mod
-      - uses: hashicorp/setup-terraform@b9cd54a3c349d3f38e8881555d616ced269862dd # v3.1.2
 
-      - name: build provider
-        run: go build .
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@b9cd54a3c349d3f38e8881555d616ced269862dd # v3.1.2
+
+      - name: Generate current schema
+        run: |
+          go build .
+          ./scripts/generate_schema.sh > current_schema.json
+
+      - name: Check for schema changes
+        id: schema-check
+        run: |
+          # Compare with committed schema if it exists
+          if [ -f "provider_schema.json" ]; then
+            if ! diff -q provider_schema.json current_schema.json > /dev/null; then
+              echo "schema_changed=true" >> $GITHUB_OUTPUT
+              echo "Schema changes detected!"
+            else
+              echo "schema_changed=false" >> $GITHUB_OUTPUT
+              echo "No schema changes detected."
+            fi
+          else
+            echo "schema_changed=true" >> $GITHUB_OUTPUT
+            echo "No existing schema file found."
```

Comment:
nit: maybe being more explicit on the expectation, here? (which is a bit different from the above `diff` case)
```
echo "Required provider_schema.json file not found"
```